src/App.test.tsx

```diff
@@ -66,9 +70,45 @@ describe('App', () => {
       <Provider store={store}>
         <App />
       </Provider>,
+      { wrapper: BrowserRouter },
     )
     await waitFor(() => {
       expect(screen.getByText('Error')).toBeInTheDocument()
     })
   })
 })
+
+describe('App | Router', () => {
+  const renderWithRouter = (ui: React.ReactElement, { route = '/' } = {}) => {
+    window.history.pushState({}, 'Test Page', route)
+    return {
+      user: userEvent,
+      ...render(ui, { wrapper: BrowserRouter }),
+    }
+  }
```

Comment:
It's better for you to abstract this function at a higher level as you have assigned `BrowserRouter` in multiple places too